### PR TITLE
ArrayImageSource admit numpy

### LINKED ISSUE #403: ArrayImageSource should admit Arrays
I think this was oversight (probably my bad) to require `ArrayImageSource` to be initialized from `Image`.

We can attempt to do this for the user (checking for sanity). It should only require a few lines of code, basic unit test, and updating our example notebooks/tutorials/docs.

Result should make our examples less terse, and a more sensible api for users.

### PR DESCRIPTION
**Pull request by j-c-c**
Adds code and tests for using numpy arrays directly with ArrayImageSource.

### CODE REVIEW

**Comment by garrettwrong** [comment]
Cool so I think in `tests/test_array_image_source.py`, we need to add a test that exercises the new branch. It's a sort of silly test, but we mainly want to ensure we didn't typo etc in a way that would make us crash.  IE, its just a smoke test.  Copied from `testArrayImageSource`.

```
    def testArrayImageSourceFromNumpy(self):
        """                                                                              
        An Array can be wrapped in an ArrayImageSource when we need to deal with ImageSource objects.                                                                            
                                                                                         
        This checks round trip conversion does not crash and returns identity.
        """

        # Create an ArrayImageSource directly from Numpy array
        src = ArrayImageSource(self.ims_np)

        # Ask the Source for all images in the stack as a Numpy array
        ims_np = src.images(start=0, num=np.inf).asnumpy()  

        # Comparison should be yield identity
        self.assertTrue(np.allclose(im, self.ims_np))
```

I have not tested this, but I think something like this should work fine for us.  @j-c-c , could you try to add this into the branch, test it, commit, and push it up?

File: src/aspire/source/image.py [hunk 1]
```diff
@@ -763,11 +763,20 @@ class ArrayImageSource(ImageSource):
     def __init__(self, im, metadata=None, angles=None):
         """
         Initialize from an `Image` object
-        :param im: An `Image` object representing image data served up by this `ImageSource`
+        :param im: An `Image` or Numpy array object representing image data served up by this `ImageSource`
```
Comment (garrettwrong) [inline src/aspire/source/image.py]: maybe missing punctuation (, or .) at end of line.

**Review by garrettwrong — CHANGES_REQUESTED** [review]
Looks good so far, thanks.  We still need to:

1. Add a test that tries to create a class with non compliant input (say a numpy array with shape (3,2,1) ), and check it returns a meaningful error message. 
 
2. Use this feature to simplify some of the tutorial/example code.

**Review by garrettwrong — APPROVED** [review]
Cool, I think this covers the issue.  Thanks.

**Review by janden — APPROVED** [review]
LGTM

**Pull request merged**